### ISSUE TITLE
Issue in the Flask application

### ISSUE THREAD

**Issue by sonjahuber1** [sid027/dt1-23#10]
The bubble app did not run after completing all the steps, 
i tested directly in the end point ( http://34.65.182.30:5000/chat?model_id=facebook/blenderbot-400M-distill&input=test ) 
and it showed an issue within the call_huggingface_chat_model function in the Flask application, specifically at the point where it tries to jsonify the response from the Hugging Face API.

To test the API I used a a curl command on my terminal, and it showed me the GTP generated answers. 

so I also got a suggestion to update the main.py file, so i guess it is about the JSON responses in Flask. 
what would you recommend? is it possible to modify the responses without changing the main.py file?

**Comment by sid027** [comment]
I do not think that you should need to update the main.py. Can you post the error please?

**Comment by e-kartofel** [comment]
Solved in Teams:
![image](https://github.com/sid027/dt1-23/assets/148684734/a2f678da-4558-4c12-85fd-54ab08d7a8ee)
Although this solution requires the user to perform a complex setting operation in their browser, it is specified like this in the Assignment specs.
![image](https://github.com/sid027/dt1-23/assets/148684734/e99b57a2-1263-4f1d-91cc-1da9847334ab)

Please @sid027 close the threat if the solution is good.

**Comment by margohasi** [comment]
@sonjahuber1 Did you solve this issue? I have the same problem with CORS Policy and I have allowed mixed content. I don't know how to solve this issue.

**Comment by sonjahuber1** [comment]
unfortunately not, @margohasi could you share if you find a solution please?

**Comment by margohasi** [comment]
> unfortunately not, @margohasi could you share if you find a solution please? Dylan helped me with it. I think it is because of Mac Book Security.  So there is a command where you can type into your local terminal to deactivate all the browser settings: /Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --disable-web-security --user-data-dir="/Users/margauxhasler/GitHub/ChromeDevSession"  -> but ask ChatGPT how to use it. Because it really exposes you to security issues:

**Comment by sid027** [comment]
#13

**Comment by joshlevent** [comment]
> <img alt="Screenshot 2023-11-10 at 13 20 27" width="624" src="https://user-images.githubusercontent.com/144933075/282057625-b43893d6-da34-4a1e-b33b-641447c85bea.png">
> also there is a problem with mixed content, even though i have the Allow CORS running and i changed security settings. (also the latest version of chrome has removed the option to allow mixed content)

I think the “mixed content” warning is not the issue. If you enabled it in your browser security settings you can ignore the yellow warning. You have two error messages: "net::ERR_CONNECTION_REFUSED” and "TypeError:”. The first is an issue with your server, possibly with the firewall not allowing the connection on the port you want. The second issue looks like a javascript error, I think the javascript code is not quite working yet. I would try to diagnose these issues separately by connection to your server using the command-line or a tool like postman, until you can solve the connection issue. Then you can focus on getting the javascript in bubble to work.
